src/nvim/ui_bridge.c

```diff
@@ -181,12 +182,19 @@ static void ui_bridge_cursor_goto_event(void **argv)
 
 static void ui_bridge_cursor_styleset(UI *b, Dictionary args)
 {
-  UI_CALL(b, cursor_styleset, 2, b, &args);
+  Object copy = copy_object(DICTIONARY_OBJ(args));
+  Object *a = xmalloc(sizeof(Object));
+  *a = copy;
+  UI_CALL(b, cursor_styleset, 2, b, a);
```

Comment:
am I right in doing a deep copy of the dictionary ? as the events can be deferred, I suppose I have to do it else we could send an freed memory in the event ?